x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.test.ts

```diff
@@ -612,4 +612,178 @@ describe('IndexPattern Data Source', () => {
       });
     });
   });
+
+  describe('#getErrorMessages', () => {
+    it('should detect a missing reference in a layer', () => {
+      const state = {
+        indexPatternRefs: [],
+        existingFields: {},
+        isFirstExistenceFetch: false,
+        indexPatterns: expectedIndexPatterns,
+        layers: {
+          first: {
+            indexPatternId: '1',
+            columnOrder: ['col1'],
+            columns: {
+              col1: {
+                dataType: 'number',
+                isBucketed: false,
+                label: 'Foo',
+                operationType: 'count', // <= invalid
+                sourceField: 'bytes',
+              },
+            },
+          },
+        },
+        currentIndexPatternId: '1',
+      };
+      const messages = indexPatternDatasource.getErrorMessages(state as IndexPatternPrivateState);
+      expect(messages).toHaveLength(1);
+      expect(messages![0]).toEqual({
+        shortMessage: 'Invalid reference',
+        longMessage: 'Field "bytes" has invalid reference',
+      });
+    });
+
+    it('should detect and batch missing references in a layer', () => {
```

Comment:
Do you have a good way to test this in the UI?

Reply:
Yes.

I used the steps in this PR to create the right environment with missing references: https://github.com/elastic/kibana/pull/78173 .

* Create a mapping for an index
```
PUT myindex
{
  "mappings": {
    "properties": {
      "myfield": {
        "type": "keyword"
      },
      "myotherfield": {
        "type": "keyword"
      }
    }
  }
}
```

* Create an indexpattern for this index
* Create a Lens visualization based on myfield (even if there is no data)
* Change the mapping
```
DELETE myindex
PUT myindex
{
  "mappings": {
    "properties": {
      "myfield": {
        "type": "text"
      },
      "myotherfield2": {
        "type": "keyword"
      }
    }
  } 
}
```
* Refresh the index pattern
* Access the visualization

Reply:
I was able to test this, and also validated that if I didn't refresh the index pattern we would get a request error.